southbound/device-factory/device-factory-core/src/main/java/org/eclipse/sensinact/gateway/southbound/device/factory/impl/FactoryParserHandler.java

```diff
@@ -913,15 +1015,14 @@ private Instant parseDateTime(String strDateTime, ZoneId timezone, String format
      */
     private Instant convertTimestamp(final String provider, final Long timestamp) {
 
-        int currentLogMs = (int) Math.log10(System.currentTimeMillis());
-        int currentLogNs = (int) Math.log10(System.nanoTime());
+        int log10ms = (int) Math.log10(System.currentTimeMillis());
         int timestampLog = (int) Math.log10(timestamp);
 
-        if (timestampLog == currentLogMs) {
+        if (timestampLog == log10ms) {
```

Comment:
I think this will have issues around base 10 rollover. Would:

    if (timestampLog > (log10ms + 2)) {
        // Over 500 years in the future - guess nanos
        return Instant.EPOCH.plusNanos(timestamp);
    } else if (timestampLog < (log10ms - 1) {
        // Over 45 years in the past, guess seconds
        return Instant.ofEpochSecond(timestamp);
    } else {
        return Instant.ofEpochMilli(timestamp);
    }

be better?

Reply:
Is it something we should handle or should we suppose the JVM will manage it?
` Instant.ofEpochSecond(Instant.MAX.getEpochSecond())` works, with Instant.MAX being end of year 1_000_000_000.

Reply:
I don't mean rollover of the epoch, just the seconds representation. If Instant.now() is `1000_000_000` seconds and the update time is `999_999_999` seconds then the `log10` won't match the `timestampLog`. I think we need slightly fuzzy matching (e.g. allowing +/-1) to allow for this scenario.

Reply:
OK I see, I'll apply the change then